processor/partition_worker.go

```diff
@@ -0,0 +1,135 @@
+package processor
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-server/jobsdb"
+	"github.com/rudderlabs/rudder-server/services/rsources"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+type partitionWorker struct {
+	partition string
+	workers   []*pipelineWorker
+	logger    logger.Logger
+	stats     *processorStats
+	handle    workerHandle
+	g         *errgroup.Group
+	ctx       context.Context
+}
+
+// newProcessorWorker creates a new worker for the specified partition
+func newPartitionWorker(ctx context.Context, partition string, h workerHandle) *partitionWorker {
```

Comment:
partition worker should use its own independent context, controlled through `Stop`
```suggestion
func newPartitionWorker(partition string, h workerHandle) *partitionWorker {
```